### PR TITLE
Update security considerations.

### PR DESCRIPTION
**Pull request by Bren2010**
Security considerations has gotten a little outdated. Updated with new info

### CODE REVIEW

**Review by kkohbrok — COMMENTED** [review]
Some very necessary updates, but I think we should be precise with regard to FS/PCS and/or leave it to the architecture document to go into detail. Especially FS (or PCFS) is very tricky in MLS.

File: draft-ietf-mls-protocol.md [hunk 1]
```diff
@@ -2668,56 +2668,47 @@ document.
...
-derived secrets.
-
-In the case where the client could have been compromised (device
-loss, for example), the client SHOULD signal the delivery service to expire
-all the previous KeyPackages and publish fresh ones for PCS.
+The first form of authentication we provide is that group members can verify a
+message originated from one of the members of the group. For encrypted messages,
+this is guaranteed because messages are encrypted with an AEAD under a key
+derived from the group secrets. For plaintext messages, this is guaranteed by
+the use of a `message_tag` which constitutes a MAC over the message, under a key
```
[the remark this answers is not in the document]
Comment (Bren2010) [reply draft-ietf-mls-protocol.md:2690]: `membership_tag` actually, https://github.com/mlswg/mls-protocol/pull/396

File: draft-ietf-mls-protocol.md [hunk 2]
```diff
@@ -2668,56 +2668,47 @@ document.
...
+derived from the group secrets.
+
+The second form of authentication is that group members can verify a message
+originated from a particular member of the group. This is guaranteed by a
+digital signature on each message from the sender's identity key.
+
+## Forward Secrecy and Post-Compromise Security
+
+Post-compromise security is provided between epochs by members regularly
```
[the remark this answers is not in the document]
Comment (Bren2010) [reply draft-ietf-mls-protocol.md:2699]: I think it's fine to only discuss PCS wrt to an individual member. FS is talked about the same way, and it makes sense to me

File: draft-ietf-mls-protocol.md [hunk 3]
```diff
@@ -2668,56 +2668,47 @@ document.
...
+updating their leaf key in the ratchet tree. Updating their leaf key prevents
+group secrets from continuing to be encrypted to previously compromised public
+keys.
+
+Forward-secrecy between epochs is provided by deleting private keys from past
+version of the ratchet tree, as this prevents old group secrets from being
+re-derived. Forward secrecy *within* an epoch is provided by deleting message
```
[the remark this answers is not in the document]
Comment (Bren2010) [reply draft-ietf-mls-protocol.md:2706]: That key material must be deleted per the protocol, so I don't feel the need to re-specify that in its entirety here

File: draft-ietf-mls-protocol.md [hunk 4]
```diff
@@ -2668,56 +2668,47 @@ document.
...
+digital signature on each message from the sender's identity key.
+
+## Forward Secrecy and Post-Compromise Security
+
+Post-compromise security is provided between epochs by members regularly
+updating their leaf key in the ratchet tree. Updating their leaf key prevents
+group secrets from continuing to be encrypted to previously compromised public
+keys.
+
+Forward-secrecy between epochs is provided by deleting private keys from past
```
[the remark this answers is not in the document]
Comment (Bren2010) [reply draft-ietf-mls-protocol.md:2704]: I think what I said is correct in principle, I'm just cutting corners. There's a disclaimer at the top of this section saying that this is just meant to give a high-level understanding of why MLS is secure but a full discussion is out of scope

File: draft-ietf-mls-protocol.md [hunk 5]
```diff
@@ -2668,56 +2668,47 @@ document.
...
 
-Group secrets are derived from (i) previous group secrets, and (ii)
-the root key of a ratcheting tree. Only group members know their leaf
-private key in the group, therefore, the root key of the group's ratcheting
-tree is secret and thus so are all values derived from it.
+Group secrets are partly derived from the output of a ratchet tree. Ratchet
+trees work by assigning each member of the group to a leaf in the tree and
+maintaining the following property: the private key of a node in the tree is
+known only to members of the group that are assigned a leaf in the node's
+subtree. This is called the *rachet tree invariant* and it makes it possible to
```
Comment (tomtau) [inline draft-ietf-mls-protocol.md:2675]:
```suggestion
subtree. This is called the *ratchet tree invariant* and it makes it possible to
```

File: draft-ietf-mls-protocol.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kkohbrok) [reply draft-ietf-mls-protocol.md:2706]: It's true that it must be deleted. It's just that deleting it doesn't give you FS within an epoch.

**Review by bifurcation — APPROVED** [review]

**Pull request merged**